packages/events-api/src/http-handler.ts

```diff
@@ -59,18 +60,20 @@ export function createHTTPHandler(adapter: SlackEventAdapter): HTTPHandler {
     return (err, responseOptions) => {
       debug('sending response - error: %s, responseOptions: %o', err, responseOptions);
       // Deal with errors up front
-      if (err !== undefined) {
+      if (!isFalsy(err)) {
         if ('status' in err) {
           res.statusCode = err.status;
-        } else if (err.code === ErrorCode.SignatureVerificationFailure ||
-            err.code === ErrorCode.RequestTimeFailure) {
+        } else if ('code' in err && (
```

Comment:
`err` here is typed as `err?: Error | CodedError | { status: ResponseStatus }`. The handlers are checked in reverse order of that union:

1. `{ status: ResponseStatus }` by checking `'status' in err`
2. *some* codes of `CodedError` by checking `'coded' in err` and if a specific code is present
3. the remaining codes of `CodedError` or `Error`
4. (`undefined` is ruled out right before this if-else chain)

This was indeed added to satisfy the typechecker because once `{ status: ... }` is eliminated, `code` may or may not exist on `Error | CodedError`. As far as I know, adding `'code' in err` to this condition should not change its truthiness.

I don't know if there's a better way to describe the type of `err` nor what it would be 🤷‍♀.

Reply:
FYI i was browsing around and saw this... a `ResponseHandler` below is `Error | CodedError | {status: number}` but that isn't quite right. You shouldn't be specifying the redundant type here, it should really be `Error | {status: number}` because a `CodedError` _is_ an `Error`. 

Because you already have a union there, you already have to do a check to figure out if its a coded error, making its involvement in the type redundant AFAIK.

Reply:
If I was refactoring this code, I'd agree! But one of my goals was to try and touch as little functionality as possible and simply add types. But this is definitely a good point, and I think it would ultimately be the right way to do this 👍

I also think there's some value in having `CodedError` in the type signature to suggest there is different functionality for (some) `CodedError`s that is different than for `Error`s. Yeah or no?

Reply:
in an ideal world i think it'd actually be just `Error`. you'd usually do this so any type of `Error` can be thrown, even ones the handler may not support (but at least understands the base interface).

it should be up to the handler to decide if it wants to handle specific kinds of `Error`.

similar to how DOM event handlers work (most take an `Error` unless they are known events which _always_ fire a specific kind of error, e.g. Click).

Reply:
I agree with @43081j on this one. The intention of the `ResponseHandler` type is to represent a function that the user can provide any `Error` as the first argument, and the specific behavior that occurs when subtypes are provided is up to the library.

I do think that the documentation should describe that behavior. I think we can add a little more description to L106 to make this clearer.

In fact, I think we might want to go a little further and take `| { status: ResponseStatus }` out of the type, but I'll reserve that discussion for [another part of the review](https://github.com/slackapi/node-slack-sdk/pull/799/files#r303687812).